aim/tests/unit/aim_lib/test_nat_strategy.py

```diff
@@ -310,6 +310,31 @@ def test_connect_vrfs(self):
         self.ns.disconnect_vrf(self.ctx, ext_net, vrf2)
         self._check_connect_vrfs('stage4')
 
+    @base.requires(['foreign_keys'])
+    def test_set_unset_bd_l3out(self):
```

Comment:
The test here will also test that its a no-op for other Nat strategies. I think its good that way because in our MD we will then invoke this set_bd_l3out() call regardless which Nat strategy its.

Reply:
In that case, define a _check_xxxx() method that is overridden in the child classes. See other tests for examples.